### ISSUE TITLE
Experimental Parts do not have the Blue icon on them

### ISSUE THREAD

**Issue by 123214523** [KSP-RO/RP-0#1882]
As said in title there no way to differ between part that have completed reasearch and parts that have not

**Comment by siimav** [comment]
Squad broke this in one of the later updates. Unfortunately no idea how they did it or how to fix it.

**Comment by talchas** [comment]
I have a fix at https://github.com/talchas/KSPCommunityFixes/, but I'm currently trying to track down what is causing some items to be stuck as locked, which I thought might have been caused by this but seems to predate trying to do this.

Also it doesn't do quite what you want, as it just highlights everything added in the same color, both "because you didn't buy it" and "because you don't have it researched at all". It does show how RP-1/KCT could highlight it with the extra data though. (The current version is a inefficient, as those things I patch actually run every frame and exit early, but the patch doesn't get skipped by that early exit; I don't know if it's actually a relevant perf cost though)

**Comment by talchas** [comment]
And here's a RP-0 patch for highlighting just unresearched ones
```csharp
        [HarmonyPatch(typeof(EditorPartIcon))]
        internal class PatchEditorPartButtons
        {
            [HarmonyPrefix]
            [HarmonyPatch("Update")]
            internal static void PatchUpdate(EditorPartIcon __instance, ref bool ___checkedExperimental)
            {
                // stock's experimental display is broken and we want to replace it anyways, so use its state variables
                if (___checkedExperimental)
                    return;

                var part = __instance.AvailPart;
                var to = __instance.gameObject.GetChild("ButtonImage")?.GetComponent<UnityEngine.UI.Image>();

                if (to == null || part == null)
                    return;

                if (KCTGameStates.TechList.Any(tech => tech.TechID == part.TechRequired))
                    to.color = __instance.experimentalPartColor;

                ___checkedExperimental = true;
            }
        }
```

**Comment by siimav** [comment]
Thanks, implemented the patch in 0fd09cddb3862494d16bf2e17a545659d6749a31
